Example SDL2 does not compile with g++ 7

Example sdl2.cpp does not compile using g++ (GCC) 7.1.1 20170630.

Output:
`g++ example/sdl2.cpp -std=c++1y -Wall -Wextra -Werror -pedantic -pedantic-errors -I include -I . -o example/sdl2.out &&  example/sdl2.out
In file included from example/sdl2.cpp:8:0:
include/boost/sml.hpp: In instantiation of ‘struct boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings’:
include/boost/sml.hpp:917:100:   required from ‘struct boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>’
include/boost/sml.hpp:896:63:   required from ‘struct boost::sml::v1_1_0::back::get_state_mapping<boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>, boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>, boost::sml::v1_1_0::aux::integral_constant<bool, false> >’
include/boost/sml.hpp:907:88:   required by substitution of ‘template<class T, class TMappings, class TUnexpected> using get_state_mapping_t = typename boost::sml::v1_1_0::back::get_state_mapping::type [with T = boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>; TMappings = boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>; TUnexpected = boost::sml::v1_1_0::aux::integral_constant<bool, false>::type]’
include/boost/sml.hpp:1105:9:   required from ‘bool boost::sml::v1_1_0::back::sm_impl< <template-parameter-1-1> >::process_event_impl(const TEvent&, TDeps&, TSubs&, const boost::sml::v1_1_0::aux::type_list<TStates ...>&, boost::sml::v1_1_0::aux::index_sequence<Ns ...>) [with TMappings = boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>; TEvent = boost::sml::v1_1_0::back::anonymous; TDeps = boost::sml::v1_1_0::aux::pool<>; TSubs = boost::sml::v1_1_0::aux::pool<boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> > >; TStates = {boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>, boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', '_', 'f', 'o', 'r', '_', 'u', 's', 'e', 'r', '_', 'i', 'n', 'p', 'u', 't'>, boost::sml::v1_1_0::aux::string<char, 'k', 'e', 'y', '_', 'p', 'r', 'e', 's', 's', 'e', 'd'>, boost::sml::v1_1_0::back::terminate_state, boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', 'i', 'n', 'g', '_', 'f', 'o', 'r', '_', 'q', 'u', 'i', 't'>}; int ...Ns = {0, 1}; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>]’
include/boost/sml.hpp:1125:41:   required from ‘bool boost::sml::v1_1_0::back::sm_impl< <template-parameter-1-1> >::process_event_noexcept(const TEvent&, TDeps&, TSubs&, boost::sml::v1_1_0::aux::false_type) [with TMappings = boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>; TEvent = boost::sml::v1_1_0::back::anonymous; TDeps = boost::sml::v1_1_0::aux::pool<>; TSubs = boost::sml::v1_1_0::aux::pool<boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> > >; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>; boost::sml::v1_1_0::aux::false_type = boost::sml::v1_1_0::aux::integral_constant<bool, false>]’
include/boost/sml.hpp:1045:88:   required from ‘bool boost::sml::v1_1_0::back::sm_impl< <template-parameter-1-1> >::process_internal_events(const TEvent&, TDeps&, TSubs&) [with TEvent = boost::sml::v1_1_0::back::anonymous; TDeps = boost::sml::v1_1_0::aux::pool<>; TSubs = boost::sml::v1_1_0::aux::pool<boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> > >; typename boost::sml::v1_1_0::aux::enable_if<(boost::sml::v1_1_0::aux::integral_constant<bool, __is_base_of(typename boost::sml::v1_1_0::back::event_type<TEvent>::generic_t, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::aux::inherit, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::aux::unique_t, typename boost::sml::v1_1_0::aux::join<typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::back::get_sub_internal_events, decltype((declval<typename TSM::sm>)().operator()())>::type, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::back::get_events, decltype((declval<typename TSM::sm>)().operator()())>::type>::type>::type>::type)>::value && (! boost::sml::v1_1_0::aux::integral_constant<bool, __is_base_of(typename boost::sml::v1_1_0::back::event_type<TEvent>::mapped_t, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::aux::inherit, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::aux::unique_t, typename boost::sml::v1_1_0::aux::join<typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::back::get_sub_internal_events, decltype((declval<typename TSM::sm>)().operator()())>::type, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::back::get_events, decltype((declval<typename TSM::sm>)().operator()())>::type>::type>::type>::type)>::value)), int>::type <anonymous> = 0; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>]’
include/boost/sml.hpp:1028:28:   required from ‘void boost::sml::v1_1_0::back::sm_impl< <template-parameter-1-1> >::start(TDeps&, TSubs&) [with TDeps = boost::sml::v1_1_0::aux::pool<>; TSubs = boost::sml::v1_1_0::aux::pool<boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> > >; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>]’
include/boost/sml.hpp:1267:5:   required from ‘boost::sml::v1_1_0::back::sm< <template-parameter-1-1> >::sm(TDeps&& ...) [with TDeps = {}; typename boost::sml::v1_1_0::aux::enable_if<boost::sml::v1_1_0::aux::is_unique<boost::sml::v1_1_0::aux::none_type, TDeps ...>::value, int>::type <anonymous> = 0; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>]’
include/boost/sml.hpp:1670:8:   required from here
include/boost/sml.hpp:987:10: error: ‘boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings’ has a base ‘boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::event_mappings<sdl_event_impl<(SDL_EventType)256>, boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::state_mappings<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', 'i', 'n', 'g', '_', 'f', 'o', 'r', '_', 'q', 'u', 'i', 't'>, boost::sml::v1_1_0::aux::type_list<boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::back::terminate_state>, boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', 'i', 'n', 'g', '_', 'f', 'o', 'r', '_', 'q', 'u', 'i', 't'>(boost::sml::v1_1_0::front::initial_state)>, boost::sml::v1_1_0::front::transition_ea<boost::sml::v1_1_0::front::event<sdl_event_impl<(SDL_EventType)256> >, boost::sml::v1_1_0::aux::zero_wrapper<sdl2::operator()() const::<lambda()>, void> > > > > > > >, boost::sml::v1_1_0::back::event_mappings<sdl_event_impl<(SDL_EventType)1026>, boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::state_mappings<boost::sml::v1_1_0::aux::string<char, 'k', 'e', 'y', '_', 'p', 'r', 'e', 's', 's', 'e', 'd'>, boost::sml::v1_1_0::aux::type_list<boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::back::terminate_state>, boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'k', 'e', 'y', '_', 'p', 'r', 'e', 's', 's', 'e', 'd'> >, boost::sml::v1_1_0::front::transition_ea<boost::sml::v1_1_0::front::event<sdl_event_impl<(SDL_EventType)1026> >, boost::sml::v1_1_0::aux::zero_wrapper<sdl2::operator()() const::<lambda()>, void> > > > > > > >, boost::sml::v1_1_0::back::event_mappings<sdl_event_impl<(SDL_EventType)769>, boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::state_mappings<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', '_', 'f', 'o', 'r', '_', 'u', 's', 'e', 'r', '_', 'i', 'n', 'p', 'u', 't'>, boost::sml::v1_1_0::aux::type_list<boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'k', 'e', 'y', '_', 'p', 'r', 'e', 's', 's', 'e', 'd'> >, boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', '_', 'f', 'o', 'r', '_', 'u', 's', 'e', 'r', '_', 'i', 'n', 'p', 'u', 't'> >, boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::event<sdl_event_impl<(SDL_EventType)769> >, boost::sml::v1_1_0::aux::zero_wrapper<<lambda(auto:3)> [with auto:3 = <unnamed enum>]::<lambda(auto:4)>, void>, boost::sml::v1_1_0::aux::zero_wrapper<sdl2::operator()() const::<lambda()>, void> > > > > > > >, boost::sml::v1_1_0::back::event_mappings<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::state_mappings<boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>, boost::sml::v1_1_0::aux::type_list<boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', '_', 'f', 'o', 'r', '_', 'u', 's', 'e', 'r', '_', 'i', 'n', 'p', 'u', 't'> >, boost::sml::v1_1_0::front::transition_sa<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>(boost::sml::v1_1_0::front::initial_state)>, boost::sml::v1_1_0::aux::zero_wrapper<sdl2::operator()() const::<lambda()>, void> > > > > > > >’ whose type uses the anonymous namespace [-Werror=subobject-linkage]
   struct mappings : mappings_t<transitions_t> {};
          ^~~~~~~~
`

I took the liberty to reformat your output (and add the important last message).
```
g++ example/sdl2.cpp -std=c++1y -Wall -Wextra -Werror -pedantic -pedantic-errors -I include -I . -o example/sdl2.out &&  example/sdl2.out
In file included from example/sdl2.cpp:8:0:
include/boost/sml.hpp: In instantiation of ‘struct boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings’:
include/boost/sml.hpp:917:100:   required from ‘struct boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>’
include/boost/sml.hpp:896:63:   required from ‘struct boost::sml::v1_1_0::back::get_state_mapping<boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>, boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>, boost::sml::v1_1_0::aux::integral_constant<bool, false> >’
include/boost/sml.hpp:907:88:   required by substitution of ‘template<class T, class TMappings, class TUnexpected> using get_state_mapping_t = typename boost::sml::v1_1_0::back::get_state_mapping::type [with T = boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>; TMappings = boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>; TUnexpected = boost::sml::v1_1_0::aux::integral_constant<bool, false>::type]’
include/boost/sml.hpp:1105:9:   required from ‘bool boost::sml::v1_1_0::back::sm_impl< <template-parameter-1-1> >::process_event_impl(const TEvent&, TDeps&, TSubs&, const boost::sml::v1_1_0::aux::type_list<TStates ...>&, boost::sml::v1_1_0::aux::index_sequence<Ns ...>) [with TMappings = boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>; TEvent = boost::sml::v1_1_0::back::anonymous; TDeps = boost::sml::v1_1_0::aux::pool<>; TSubs = boost::sml::v1_1_0::aux::pool<boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> > >; TStates = {boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>, boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', '_', 'f', 'o', 'r', '_', 'u', 's', 'e', 'r', '_', 'i', 'n', 'p', 'u', 't'>, boost::sml::v1_1_0::aux::string<char, 'k', 'e', 'y', '_', 'p', 'r', 'e', 's', 's', 'e', 'd'>, boost::sml::v1_1_0::back::terminate_state, boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', 'i', 'n', 'g', '_', 'f', 'o', 'r', '_', 'q', 'u', 'i', 't'>}; int ...Ns = {0, 1}; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>]’
include/boost/sml.hpp:1125:41:   required from ‘bool boost::sml::v1_1_0::back::sm_impl< <template-parameter-1-1> >::process_event_noexcept(const TEvent&, TDeps&, TSubs&, boost::sml::v1_1_0::aux::false_type) [with TMappings = boost::sml::v1_1_0::back::get_event_mapping_impl_helper<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings>; TEvent = boost::sml::v1_1_0::back::anonymous; TDeps = boost::sml::v1_1_0::aux::pool<>; TSubs = boost::sml::v1_1_0::aux::pool<boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> > >; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>; boost::sml::v1_1_0::aux::false_type = boost::sml::v1_1_0::aux::integral_constant<bool, false>]’
include/boost/sml.hpp:1045:88:   required from ‘bool boost::sml::v1_1_0::back::sm_impl< <template-parameter-1-1> >::process_internal_events(const TEvent&, TDeps&, TSubs&) [with TEvent = boost::sml::v1_1_0::back::anonymous; TDeps = boost::sml::v1_1_0::aux::pool<>; TSubs = boost::sml::v1_1_0::aux::pool<boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> > >; typename boost::sml::v1_1_0::aux::enable_if<(boost::sml::v1_1_0::aux::integral_constant<bool, __is_base_of(typename boost::sml::v1_1_0::back::event_type<TEvent>::generic_t, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::aux::inherit, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::aux::unique_t, typename boost::sml::v1_1_0::aux::join<typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::back::get_sub_internal_events, decltype((declval<typename TSM::sm>)().operator()())>::type, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::back::get_events, decltype((declval<typename TSM::sm>)().operator()())>::type>::type>::type>::type)>::value && (! boost::sml::v1_1_0::aux::integral_constant<bool, __is_base_of(typename boost::sml::v1_1_0::back::event_type<TEvent>::mapped_t, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::aux::inherit, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::aux::unique_t, typename boost::sml::v1_1_0::aux::join<typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::back::get_sub_internal_events, decltype((declval<typename TSM::sm>)().operator()())>::type, typename boost::sml::v1_1_0::aux::apply<boost::sml::v1_1_0::back::get_events, decltype((declval<typename TSM::sm>)().operator()())>::type>::type>::type>::type)>::value)), int>::type <anonymous> = 0; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>]’
include/boost/sml.hpp:1028:28:   required from ‘void boost::sml::v1_1_0::back::sm_impl< <template-parameter-1-1> >::start(TDeps&, TSubs&) [with TDeps = boost::sml::v1_1_0::aux::pool<>; TSubs = boost::sml::v1_1_0::aux::pool<boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> > >; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>]’
include/boost/sml.hpp:1267:5:   required from ‘boost::sml::v1_1_0::back::sm< <template-parameter-1-1> >::sm(TDeps&& ...) [with TDeps = {}; typename boost::sml::v1_1_0::aux::enable_if<boost::sml::v1_1_0::aux::is_unique<boost::sml::v1_1_0::aux::none_type, TDeps ...>::value, int>::type <anonymous> = 0; TSM = boost::sml::v1_1_0::back::sm_policy<sdl2>]’
include/boost/sml.hpp:1670:8:   required from here
include/boost/sml.hpp:987:10: error: ‘boost::sml::v1_1_0::back::sm_impl<boost::sml::v1_1_0::back::sm_policy<sdl2> >::mappings’ has a base ‘boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::event_mappings<sdl_event_impl<(SDL_EventType)1>, boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::state_mappings<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', 'i', 'n', 'g', '_', 'f', 'o', 'r', '_', 'q', 'u', 'i', 't'>, boost::sml::v1_1_0::aux::type_list<boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::back::terminate_state>, boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', 'i', 'n', 'g', '_', 'f', 'o', 'r', '_', 'q', 'u', 'i', 't'>(boost::sml::v1_1_0::front::initial_state)>, boost::sml::v1_1_0::front::transition_ea<boost::sml::v1_1_0::front::event<sdl_event_impl<(SDL_EventType)1> >, boost::sml::v1_1_0::aux::zero_wrapper<sdl2::operator()() const::<lambda()>, void> > > > > > > >, boost::sml::v1_1_0::back::event_mappings<sdl_event_impl<(SDL_EventType)3>, boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::state_mappings<boost::sml::v1_1_0::aux::string<char, 'k', 'e', 'y', '_', 'p', 'r', 'e', 's', 's', 'e', 'd'>, boost::sml::v1_1_0::aux::type_list<boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::back::terminate_state>, boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'k', 'e', 'y', '_', 'p', 'r', 'e', 's', 's', 'e', 'd'> >, boost::sml::v1_1_0::front::transition_ea<boost::sml::v1_1_0::front::event<sdl_event_impl<(SDL_EventType)3> >, boost::sml::v1_1_0::aux::zero_wrapper<sdl2::operator()() const::<lambda()>, void> > > > > > > >, boost::sml::v1_1_0::back::event_mappings<sdl_event_impl<(SDL_EventType)2>, boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::state_mappings<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', '_', 'f', 'o', 'r', '_', 'u', 's', 'e', 'r', '_', 'i', 'n', 'p', 'u', 't'>, boost::sml::v1_1_0::aux::type_list<boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'k', 'e', 'y', '_', 'p', 'r', 'e', 's', 's', 'e', 'd'> >, boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', '_', 'f', 'o', 'r', '_', 'u', 's', 'e', 'r', '_', 'i', 'n', 'p', 'u', 't'> >, boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::event<sdl_event_impl<(SDL_EventType)2> >, boost::sml::v1_1_0::aux::zero_wrapper<<lambda(auto:3)> [with auto:3 = <unnamed enum>]::<lambda(auto:4)>, void>, boost::sml::v1_1_0::aux::zero_wrapper<sdl2::operator()() const::<lambda()>, void> > > > > > > >, boost::sml::v1_1_0::back::event_mappings<boost::sml::v1_1_0::back::anonymous, boost::sml::v1_1_0::aux::inherit<boost::sml::v1_1_0::back::state_mappings<boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>, boost::sml::v1_1_0::aux::type_list<boost::sml::v1_1_0::front::transition<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'w', 'a', 'i', 't', '_', 'f', 'o', 'r', '_', 'u', 's', 'e', 'r', '_', 'i', 'n', 'p', 'u', 't'> >, boost::sml::v1_1_0::front::transition_sa<boost::sml::v1_1_0::front::state<boost::sml::v1_1_0::aux::string<char, 'i', 'd', 'l', 'e'>(boost::sml::v1_1_0::front::initial_state)>, boost::sml::v1_1_0::aux::zero_wrapper<sdl2::operator()() const::<lambda()>, void> > > > > > > >’ whose type uses the anonymous namespace [-Werror=subobject-linkage]
   struct mappings : mappings_t<transitions_t> {};
          ^~~~~~~~
cc1plus: all warnings being treated as errors
```
I think this error has a similar reason as #112 .

--- a/example/sdl2.cpp
+++ b/example/sdl2.cpp
@@ -50,7 +50,11 @@ struct sdl_event_impl {
 template <SDL_EventType Id>
 decltype(sml::event<sdl_event_impl<Id>>) sdl_event{};
 
-auto is_key = [](auto key) { return [=](auto event) { return event.data.key.keysym.sym == key; }; };
+struct IsKey {
+  auto operator()(int key) {
+    return [=](auto event) { return event.data.key.keysym.sym == key; };
+  }
+} is_key;
 
 struct sdl2 {
   auto operator()() const noexcept {

